### PR TITLE
raidboss: convert Regexes.hasHP to watchCombatant

### PR DESCRIPTION
**Pull request by quisquous**

### CODE REVIEW

**Comment by quisquous** [comment]
Needs some testing, but here's a first pass.

**Comment by valarnin** [comment]
Just a quick note that `watchCombatants` probably doesn't work 100% right yet in emulator, I haven't had time to handle that portion and it uses its own setTimeout calls so it might just stall emulator or something.

File: ui/raidboss/data/04-sb/ultimate/unending_coil_ultimate.js [hunk 1]
```diff
@@ -250,44 +257,40 @@ export default {
...
+      netRegexKo: NetRegexes.startsUsing({ id: '26A[AE]', source: '트윈타니아' }),
+      condition: (data) => !data.monitoringHP && data.hpThresholds[data.currentPhase] !== undefined,
+      preRun: (data) => data.monitoringHP = true,
+      promise: (data, matches) => Util.watchCombatant({
+        ids: [parseInt(matches.sourceId, 16)],
+      },
+      (ret) => {
+        return ret.combatants.some((c) => {
+          const currentHPCheck = data.hpThresholds[data.currentPhase] ?? -1;
+          return currentHPCheck >= (c.CurrentHP / c.MaxHP);
```
Comment (panicstevenson) [inline ui/raidboss/data/04-sb/ultimate/unending_coil_ultimate.js:277]: I know these are my doing, but to nit:
```suggestion
          return c.CurrentHP / c.MaxHP <= currentHPCheck ;
```
Just to match the new ones you added (I feel this form is more readable: "return true if any combatants have a lower HP % than the current HP check").

**Review by panicstevenson — COMMENTED** [review]
LGTM; I'll review again whenever it's out of draft mode. Let me know if you need any help testing.

**Comment by quisquous** [comment]
I still haven't tested this yet, sorry.  Too distracted with other things.

File: ui/raidboss/data/04-sb/ultimate/unending_coil_ultimate.js (hunk 1 above)
Comment (quisquous) [reply ui/raidboss/data/04-sb/ultimate/unending_coil_ultimate.js]: Done, thanks!

**Comment by quisquous** [comment]
Ok, I tested t6, t7, t9 and those all worked, so I'll call this tested enough.

**Review by panicstevenson — APPROVED** [review]

**Pull request merged**